primitives/src/valid_proof_of_investor/v2.rs

```diff
@@ -3,33 +3,33 @@ use crate::{investor_zkproof_data::v2::InvestorZKProofData, CddId, Claim, Identi
 use confidential_identity::{
     claim_proofs::{slice_to_scalar, Verifier},
     cryptography_core::cdd_claim::CddId as CryptoCddId,
-    CompressedRistretto, Scalar, VerifierTrait as _,
+    CompressedRistretto, VerifierTrait as _,
 };
 
-/// Evaluates if the claim is a valid proof.
+/// Evaluates if the claim is a valid proof and if `scope_id` matches with the one inside `proof`.
 pub fn evaluate_claim(claim: &Claim, id: &IdentityId, proof: &InvestorZKProofData) -> bool {
-    match claim {
-        Claim::InvestorUniqueness(_, scope_did, cdd_id) => {
-            verify_proof(id, scope_did, cdd_id, proof)
+    if let Claim::InvestorUniqueness(scope, scope_id, cdd_id) = claim {
+        let proof_scope_id = proof.0.scope_id.compress().to_bytes();
+        if proof_scope_id == scope_id.as_bytes() {
```

Comment:
@fmiguelgarcia I'm still not seeing why we need `scope_id` in the claim when we can just get it from `proof.0.scope_id`.

We can have a new claim variant, `Claim::InvestorUniquenessv2` or something like that that does not have `scope_id`.